xtask/src/themekeyscheck.rs

```diff
@@ -0,0 +1,430 @@
+use std::collections::BTreeMap;
+use std::fmt::Write;
+
+use helix_core::syntax::read_query;
+use helix_core::tree_sitter::Query;
+use helix_loader::grammar::get_language;
+
+use crate::helpers::lang_config;
+
+pub fn theme_keys_check() -> Result<(), String> {
+    let highlight_queries_file = "highlights.scm";
+    let mut highlights_queries: BTreeMap<_, Vec<_>> = BTreeMap::new();
+
+    for language in lang_config().language {
+        let language_name = language.language_id.to_ascii_lowercase();
+        let grammar_name = language.grammar.unwrap_or(language.language_id);
+        let language = get_language(&grammar_name);
+        let query_text = read_query(&language_name, highlight_queries_file);
+        if let (false, Ok(language)) = (query_text.is_empty(), language) {
+            match Query::new(language, &query_text) {
+                Err(reason) => {
+                    return Err(format!(
+                        "Failed to parse {} queries for {}: {}",
+                        highlight_queries_file, language_name, reason
+                    ))
+                }
```

Comment:
We might want to eliminate the query-check xtask in favor of this once since we need the queries here anyways

Reply:
I don't think so: this only checks `highlights.scm`, not the others and conflating both is probably confusing because of that: checking themes could fail on indentation queries, that's not a good experience